recipes/tcl/8.6.9/conanfile.py

```diff
@@ -0,0 +1,46 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class TclConan(ConanFile):
+    name = "tcl"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.tcl.tk/"
+    settings = "os_build", "arch_build"
+    license = "custom"
+    description = "TCL scripting language"
+    topics = ("TCL")
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os_build not in ("Windows", "Macos", "Linux"):
+            raise ConanInvalidConfiguration(f"Unsupported os_build: {self.settings.os_build}")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], keep_permissions=True)
+        extracted_dir = self.name + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        # conan-center hook recommends putting everything under the bin folder,
+        # otherwise it rejects man/share folders
+        # also, it forbids pkg-config (.pc) files
+        install_dir = os.path.join(self.build_folder, "install")
+        if self.settings.os_build == "Windows":
```

Comment:
This should be `if self.settings.compiler == "Visual Studio":`

Reply:
This would prevent MinGW users to install it though

Reply:
Does the autotools toolchain not work for mingw?
I see that the bincrafters tcl recipe has not enabled mingw builds.

This recipe does only provide a tcl executable, no libraries.
How will these be provided?
I remember tcltk needing access to those.

Reply:
I did not know there was a tcl recipe from bincrafters, I'll take a look tomorrow and will most likely add it instead.

What I mean by MinGW users, is having a profile with `compiler=gcc` but `os_build=Windows`. This is why I cannot use the compiler setting here.

Reply:
Wouldn't it be ok that these mingw users build their tcl package using mingw?
I know, this will introduce ambiguity as to what compiler has been used to build tcl on Windows.
But now it is impossible to use mingw to build tcl.

Reply:
That's a good question, I did not consider TCL libraries at first.
I think you're right and we should support building with MinGW.

One annoying thing is that `make` and associated tools are not shipped with MinGW, but require using `msys2` or an alternative.